apps/trading/e2e/tests/referrals/test_referrals.py

```diff
@@ -0,0 +1,151 @@
+    }
+
+
+def generate_referral_expected_value_dic(expected_volume, expected_tier, expected_discount, expected_epochs, expected_epochs_to_next_tier):
+    return {
+        '[data-testid=combined-volume-value]': expected_volume,
+        '[data-testid=current-tier-value]': expected_tier,
+        '[data-testid=discount-value]': expected_discount,
+        '[data-testid=epochs-in-set-value]': expected_epochs,
+        '[data-testid=epochs-to-next-tier-value]': expected_epochs_to_next_tier
+    }
+
+
+def check_tile_values(page: Page, expected_results: dict):
+    if "referrals" in page.url:
+        page.reload()
+    else:
+        page.goto("/#/referrals/")
+
+    for selector, expected_text in expected_results.items():
+        assert selector_contains_text(
+            page, selector, expected_text), f"Expected text '{expected_text}' not found in selector '{selector}'"
+
+
+def create_benefit_tier(minimum_running_notional_taker_volume, minimum_epochs, referral_reward_factor, referral_discount_factor):
+    return {
+        "minimum_running_notional_taker_volume": minimum_running_notional_taker_volume,
+        "minimum_epochs": minimum_epochs,
+        "referral_reward_factor": referral_reward_factor,
+        "referral_discount_factor": referral_discount_factor,
+    }
+
+
+def create_staking_tier(minimum_staked_tokens, referral_reward_multiplier):
+    return {
+        "minimum_staked_tokens": minimum_staked_tokens,
+        "referral_reward_multiplier": referral_reward_multiplier,
+    }
+
+
+def setup_market_and_referral_scheme(vega: VegaService, continuous_market: str, page: Page):
+    page.goto(f"/#/markets/{continuous_market}")
+
+    create_and_faucet_wallet(vega=vega, wallet=PARTY_A)
+    create_and_faucet_wallet(vega=vega, wallet=PARTY_B)
+    forward_time(vega)
+
+    benefit_tiers = []
+    staking_tiers = []
+    for i in range(1, 4):
+        benefit_tiers.append(create_benefit_tier(
+            i * 100, i, i * 0.01, i * 0.01))
+        staking_tiers.append(create_staking_tier(
+            i * 100, i))
+
+    vega.update_referral_program(
+        proposal_key=MM_WALLET.name,
+        benefit_tiers=benefit_tiers,
+        staking_tiers=staking_tiers,
+        window_length=1,
+    )
+    forward_time(vega, True)
+
+    vega.create_referral_set(key_name=PARTY_A.name)
+    forward_time(vega, True)
+
+    referral_set_id = list(vega.list_referral_sets().keys())[0]
+    vega.apply_referral_code(key_name=PARTY_B.name, id=referral_set_id)
+
+    tdai_id = vega.find_asset_id(symbol="tDAI")
+    vega.mint(
+        "Key 1",
+        asset=tdai_id,
+        amount=10e6,
+    )
+    vega.mint(
+        PARTY_B.name,
+        asset=tdai_id,
+        amount=10e6,
+    )
+
+    submit_liquidity(vega, MM_WALLET.name, continuous_market, 100, 100)
+    forward_time(vega)
+
+
+@pytest.mark.usefixtures("page", "auth", "risk_accepted")
+def test_referral_scenario(continuous_market, vega: VegaService, page: Page):
```

Comment:
No blocking, but pulling this test out into separate tests could be good for the future.